sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -467,10 +494,170 @@ The `node` class provides the {crs}[common reference semantics].
+dynamic parameter will be reflected in the modifiable graph which contains this
+node. These updated nodes can then be passed to an executable graph to update
+it with new values.
+
+The type of the underlying object a dynamic parameter represents is set at
+compile time using a template parameter. This underlying type can be an
+accessor, a pointer to a USM allocation, scalar passed by value, or a raw byte
+representation of the argument. The raw byte representation is intended to
+enable updating arguments set using
+link:../proposed/sycl_ext_oneapi_raw_kernel_arg.asciidoc[sycl_ext_oneapi_raw_kernel_arg].
+
+Dynamic parameters are registered with nodes in a modifiable graph, with each
+registration associating one or more node arguments to the dynamic parameter
+instance. Registration happens inside the command-group that the node represents
+, and is done using the argument index (known when argument originally set using
+`set_arg()`/`set_args()`). It is valid for a node argument to be registered with
+more than one dynamic parameter instance.
+
+Registration is done by using the argument index (known when argument
+originally set using `set_arg()`/`set_args()`).
+
+See <<executable-graph-update, Executable Graph Update>> for more information
+about updating node parameters.
+
+The `dynamic_parameter` class provides the {crs}[common reference semantics].
+
+Table {counter: tableNumber}. Member functions of the `dynamic_parameter` class.
+[cols="2a,a"]
+|===
+|Member Function|Description
+
+|
+[source,c++]
+----
+dynamic_parameter(command_graph<graph_state::modifiable> graph);
+----
+|Constructs a dynamic parameter object that can be registered with node
+arguments.
+
+Parameters:
+
+* `graph` - Graph containing the nodes which will have dynamic parameters.
+
+|
+[source,c++]
+----
+dynamic_parameter(const ValueT &initialValue, command_graph<graph_state::modifiable> graph);
+----
+|Constructs a dynamic parameter object that can be registered with node
+arguments, while providing an intiial value which is used solely for
+deducing ValueT.
+
+Parameters:
+
+* `initialValue` - Initial value used for CTAD.
+* `graph` - Graph containing the nodes which will have dynamic parameters.
+
+|
+[source,c++]
+----
+void register_with_node(const handler &cgh, int argIndex);
+----
+|Callable only from command-group scope when adding a node to a graph
+using the explicit API. Associate the dynamic parameter
+with the kernel parameter of the node that encapsulates this command-group.
+Parameters are associated by their index which was given when they were
+registered with `handler`.
+
+Parameters:
+
+* `handler` - SYCL handler object.
+* `argIndex` - Kernel argument index starting from zero to associate with
+  the dynamic parameter.
+
+Exceptions:
+
+* Throws with error code `invalid` if `argIndex` is not a valid argument index
+  for the resulting node.
+* Throws with error code `invalid` if the type of the command-group is not a
+  kernel execution.
```

Comment:
This is not implementable I think, since most likely this function is called before the `parallel_for` etc. are called on the `handler`, the `handler` is not aware of what type of `node` it will produce.

We can throw this error during `handler::finalize()` so perhaps this error can be moved to the `graph::add()` definition.